products/views.py

```diff
@@ -1,3 +1,73 @@
-from django.shortcuts import render
+import json
 
-# Create your views here.
+from django.http    import JsonResponse
+from django.views   import View
+
+from products.models import Categories, Images, Products, Sale, Review
+from orders.models   import Options
+
+class ProductListView(View):
+    def get(self,request):
+        category    = request.GET.get('name', None)
+        products    = Products.objects.all()
+
+        if category=="ourfoods":
+            treats = Categories.objects.get(name="treats")
+            products = Products.objects.exclude(category_id = treats)
+
+        else:
+            category_id    = Categories.objects.get(name=category).id
+            products    = Products.objects.filter(category_id=category_id)
+
+        products_list = [{
+        "id"            : products[i].id,
+        "name"          : products[i].name,
+        "main_image"    : Products.objects.get(id = products[i].id).images_set.all().first().image_url,
+        "sub_image"     : Products.objects.get(id = products[i].id).images_set.all().last().image_url,
+        "category"      : products[i].category.name,
+        "price"         : products[i].price,
+        "stock"         : products[i].stock_status,
+        "sale_rate"     : products[i].sale_rate.sale_rate} for i in range(len(products))]
```

Comment:
아래와 같이 range를 사용하지 않고 구현해주세요!
```suggestion
        "sale_rate"     : products[i].sale_rate.sale_rate} for i in range(len(products))]
```

Reply:
수정완료하엿습니다.